libs/jwst-codec/src/doc/store.rs

```diff
@@ -361,6 +378,72 @@ impl DocStore {
             delete_set.add(id.client, id.clock, len)
         }
     }
+
+    fn diff_state_vectors(
+        local_state_vector: &StateVector,
+        remote_state_vector: &StateVector,
+    ) -> Vec<(Client, Clock)> {
+        let mut diff = Vec::new();
+
+        for (client, &remote_clock) in remote_state_vector.iter() {
+            let local_clock = local_state_vector.get(client);
+            if local_clock > remote_clock {
+                diff.push((*client, remote_clock));
+            }
+        }
+
+        for (client, _) in local_state_vector.iter() {
+            if remote_state_vector.get(client) == 0 {
+                diff.push((*client, 0));
+            }
+        }
+
+        diff
+    }
+
+    pub fn encode_with_state_vector<W: CrdtWriter>(
```

Comment:
consider assemble state as update? update has implemented the write operation: https://github.com/toeverything/OctoBase/blob/bc63b9a92d77e3dca1d77ecb2c93949cb1776690/libs/jwst-codec/src/doc/codec/update.rs#L58-L83